pinot-query-runtime/src/main/java/org/apache/pinot/query/mailbox/channel/MailboxContentObserver.java

```diff
@@ -58,10 +61,18 @@ public void onNext(MailboxContent mailboxContent) {
     if (_mailbox == null) {
       _mailbox = _mailboxService.getReceivingMailbox(mailboxId);
     }
+    if (_mailboxBuffers == null) {
+      _mailboxBuffers = new ArrayList<>();
+    }
```

Comment:
nit: What is the advantage of this lazy initialization instead of doing it eagerly?

Reply:
tip: ArrayLists do not allocate the array on construction unless you specify a custom size (see https://github.com/AdoptOpenJDK/openjdk-jdk11/blob/19fb8f93c59dfd791f62d41f332db9e306bc1422/src/java.base/share/classes/java/util/ArrayList.java#L166)

Reply:
Nice one. I always assumed it just used a default capacity, but that makes a lot more sense.